diff -up guile-3.0.7/test-suite/standalone/Makefile.am.4~ guile-3.0.7/test-suite/standalone/Makefile.am
--- guile-3.0.7/test-suite/standalone/Makefile.am.4~	2021-05-10 10:07:12.000000000 +0200
+++ guile-3.0.7/test-suite/standalone/Makefile.am	2021-08-09 18:48:13.980515045 +0200
@@ -298,7 +298,4 @@ TESTS += test-smob-mark-race
 check_SCRIPTS += test-stack-overflow
 TESTS += test-stack-overflow
 
-check_SCRIPTS += test-out-of-memory
-TESTS += test-out-of-memory
-
 EXTRA_DIST += ${check_SCRIPTS}
diff -up guile-3.0.7/test-suite/standalone/Makefile.in.4~ guile-3.0.7/test-suite/standalone/Makefile.in
--- guile-3.0.7/test-suite/standalone/Makefile.in.4~	2021-05-10 10:30:31.000000000 +0200
+++ guile-3.0.7/test-suite/standalone/Makefile.in	2021-08-09 18:49:14.053742788 +0200
@@ -106,7 +106,7 @@ TESTS = test-system-cmds test-bad-identi
 	test-scm-c-bind-keyword-arguments$(EXEEXT) \
 	test-srfi-4$(EXEEXT) $(am__append_6) $(am__EXEEXT_1) \
 	test-smob-mark$(EXEEXT) test-smob-mark-race$(EXEEXT) \
-	test-stack-overflow test-out-of-memory
+	test-stack-overflow
 check_PROGRAMS = test-num2integral$(EXEEXT) test-round$(EXEEXT) \
 	test-foreign-object-c$(EXEEXT) test-list$(EXEEXT) \
 	test-unwind$(EXEEXT) test-conversion$(EXEEXT) \
@@ -2147,7 +2147,7 @@ check_SCRIPTS = test-system-cmds test-ba
 	test-language test-guild-compile test-signal-fork \
 	$(am__append_3) test-foreign-object-scm test-fast-slot-ref \
 	test-mb-regexp test-use-srfi $(am__append_5) \
-	test-stack-overflow test-out-of-memory
+	test-stack-overflow
 BUILT_SOURCES = $(am__append_2)
 EXTRA_DIST = test-import-order-a.scm test-import-order-b.scm \
 	test-import-order-c.scm test-import-order-d.scm \
