diff -up guile-3.0.9/test-suite/standalone/Makefile.am.4~ guile-3.0.9/test-suite/standalone/Makefile.am
--- guile-3.0.9/test-suite/standalone/Makefile.am.4~	2023-11-21 22:30:27.174272650 +0100
+++ guile-3.0.9/test-suite/standalone/Makefile.am	2023-11-21 22:32:31.754571893 +0100
@@ -298,9 +298,6 @@ TESTS += test-smob-mark-race
 check_SCRIPTS += test-stack-overflow
 TESTS += test-stack-overflow
 
-check_SCRIPTS += test-out-of-memory
-TESTS += test-out-of-memory
-
 check_SCRIPTS += test-close-on-exec
 TESTS += test-close-on-exec
 
diff -up guile-3.0.9/test-suite/standalone/Makefile.in.4~ guile-3.0.9/test-suite/standalone/Makefile.in
--- guile-3.0.9/test-suite/standalone/Makefile.in.4~	2023-11-21 22:30:27.174272650 +0100
+++ guile-3.0.9/test-suite/standalone/Makefile.in	2023-11-21 22:32:44.568705242 +0100
@@ -106,7 +106,7 @@ TESTS = test-system-cmds test-bad-identi
 	test-scm-c-bind-keyword-arguments$(EXEEXT) \
 	test-srfi-4$(EXEEXT) $(am__append_6) $(am__EXEEXT_1) \
 	test-smob-mark$(EXEEXT) test-smob-mark-race$(EXEEXT) \
-	test-stack-overflow test-out-of-memory test-close-on-exec
+	test-stack-overflow test-close-on-exec
 check_PROGRAMS = test-num2integral$(EXEEXT) test-round$(EXEEXT) \
 	test-foreign-object-c$(EXEEXT) test-list$(EXEEXT) \
 	test-unwind$(EXEEXT) test-conversion$(EXEEXT) \
@@ -2245,7 +2245,7 @@ check_SCRIPTS = test-system-cmds test-ba
 	test-language test-guild-compile test-signal-fork \
 	$(am__append_3) test-foreign-object-scm test-fast-slot-ref \
 	test-mb-regexp test-use-srfi $(am__append_5) \
-	test-stack-overflow test-out-of-memory test-close-on-exec
+	test-stack-overflow test-close-on-exec
 BUILT_SOURCES = $(am__append_2)
 EXTRA_DIST = test-import-order-a.scm test-import-order-b.scm \
 	test-import-order-c.scm test-import-order-d.scm \
