diff -up guile-3.0.8/config.h.in.omv~ guile-3.0.8/config.h.in
--- guile-3.0.8/config.h.in.omv~	2022-08-27 23:25:15.443730511 +0200
+++ guile-3.0.8/config.h.in	2022-08-27 23:27:48.620909969 +0200
@@ -2480,7 +2480,7 @@ License along with Guile.  If not, see
    return pointers that can be freed via 'free'; it can be used
    only after including stdlib.h.  These macros cannot be used on
    inline functions.  */
-#if _GL_GNUC_PREREQ (11, 0)
+#if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 # define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 #else
 # define _GL_ATTRIBUTE_DEALLOC(f, i)
diff -up guile-3.0.8/m4/gnulib-common.m4.omv~ guile-3.0.8/m4/gnulib-common.m4
--- guile-3.0.8/m4/gnulib-common.m4.omv~	2022-08-27 23:25:35.193844071 +0200
+++ guile-3.0.8/m4/gnulib-common.m4	2022-08-27 23:26:28.890702322 +0200
@@ -154,7 +154,7 @@ AC_DEFUN([gl_COMMON_BODY], [
    return pointers that can be freed via 'free'; it can be used
    only after including stdlib.h.  These macros cannot be used on
    inline functions.  */
-#if _GL_GNUC_PREREQ (11, 0)
+#if _GL_GNUC_PREREQ (11, 0) && !defined(__clang__)
 # define _GL_ATTRIBUTE_DEALLOC(f, i) __attribute__ ((__malloc__ (f, i)))
 #else
 # define _GL_ATTRIBUTE_DEALLOC(f, i)
